stdlib/asm/crypto/dsa/rpo_falcon512.masm

```diff
@@ -0,0 +1,654 @@
+use.std::crypto::stark::deep_queries
+
+# CONSTANTS
+# =================================================================================================
+
+const.J=77321994752
+const.SQUARE_NORM_BOUND=34034726
+
+# MODULAR REDUCTION FALCON PRIME  
+# =============================================================================================
+
+#! Given dividend ( i.e. field element a ) on stack top, this routine computes c = a % 12289
+#!
+#! Expected stack state
+#!
+#! [a, ...]
+#!
+#! Output stack state looks like
+#!
+#! [c, ...] | c = a % 12289
+export.mod_12289
+    u32split
+    push.12289.0
```

Comment:
minor nit (fine to ignore): should we also have a constant for $12289$? Maybe something as simple as `M`?

Reply:
Done!